drivers/misc/nxp_lp_flexcomm/nxp_lp_flexcomm.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lp_flexcomm
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "nxp_lp_flexcomm.h"
+
+LOG_MODULE_REGISTER(nxp_lp_flexcomm, LOG_LEVEL_ERR);
+
+struct nxp_lp_flexcomm_child {
+	const struct device *dev;
+	uint8_t periph;
+	child_isr_t lp_flexcomm_child_isr;
+};
+
+struct nxp_lp_flexcomm_data {
+	struct nxp_lp_flexcomm_child *children;
+	size_t num_children;
```

Comment:
seems that an array with a defined size can be described within this struct instead of adding a pointer and a size

Reply:
I would prefer we keep the current implementation.